### PR TITLE
Add division operators for Matx.

### PR DESCRIPTION
**Pull request by catree**
Add the following operations for `Matx`:

<cut/>

```
    {
        Matx31d P1(1, 2, 3);
        P1 /= 2;
        std::cout << "P1: " << P1.t() << std::endl;
    }
    {
        Matx31d P1(1, 2, 3);
        P1 /= 2.0f;
        std::cout << "P1: " << P1.t() << std::endl;
    }
    {
        Matx31d P1(1, 2, 3);
        P1 /= 2.0;
        std::cout << "P1: " << P1.t() << std::endl;
    }
    {
        Matx31d P1(1, 2, 3);
        P1 = P1 / 2;
        std::cout << "P1: " << P1.t() << std::endl;
    }
    {
        Matx31d P1(1, 2, 3);
        P1 = P1 / 2.0f;
        std::cout << "P1: " << P1.t() << std::endl;
    }
    {
        Matx31d P1(1, 2, 3);
        P1 = P1 / 2.0;
        std::cout << "P1: " << P1.t() << std::endl;
    }
    {
        Matx31d P1(1, 2, 3);
        std::cout << "P1: " << (P1 / 2).t() << std::endl;
    }
    {
        Matx31d P1(1, 2, 3);
        std::cout << "P1: " << (P1 / 2.0f).t() << std::endl;
    }
    {
        Matx31d P1(1, 2, 3);
        std::cout << "P1: " << (P1 / 2.0).t() << std::endl;
    }
```

Output:

```
P1: [0.5, 1, 1.5]
P1: [0.5, 1, 1.5]
P1: [0.5, 1, 1.5]
P1: [0.5, 1, 1.5]
P1: [0.5, 1, 1.5]
P1: [0.5, 1, 1.5]
P1: [0.5, 1, 1.5]
P1: [0.5, 1, 1.5]
P1: [0.5, 1, 1.5]
```

---

TODO:
- [ ] Add unit tests (point me to the file where I can add the corresponding tests)

### CODE REVIEW

File: modules/core/include/opencv2/core/matx.hpp [hunk 1]
```diff
@@ -1275,6 +1275,48 @@ Matx<_Tp, m, n> operator * (double alpha, const Matx<_Tp, m, n>& a)
     return Matx<_Tp, m, n>(a, alpha, Matx_ScaleOp());
 }
 
+template<typename _Tp, int m, int n> static inline
+Matx<_Tp, m, n>& operator /= (Matx<_Tp, m, n>& a, int alpha)
```
Comment (alalek) [inline modules/core/include/opencv2/core/matx.hpp]: Matx usually makes sense (used) with floating point numbers (float/double).
So "int" scalar looks out of scope here. Do we really need it?

For floating point computations we can replace multiple divisions by multiplications (they are still more effective than divisions):
```
_Tp alpha_inv = (_Tp)1.0f / alpha;
for( int i = 0; i < m*n; i++ )
        a.val[i] = a.val[i] * alpha_inv;
```

`saturate_cast()` is not used for floating point numbers too.

**Comment by VadimLevin** [comment]
Things to be done to complete this PR:
 
- [ ] Leave only floating point part of the added functionality (Avoid `saturate_cast` usage for floating point types)
- [ ] Add unit tests (test file: `opencv/modules/core/test/test_operations.cpp`)

**Comment by asmorkalov** [comment]
@alalek I finalized the PR as it was discussed on the meeting. Please take a look and merge.
